.github/workflows/tests.yml

```diff
@@ -48,11 +49,20 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-${{ matrix.python-version }}-${{ env.cache-name }}-${{ hashFiles('**/setup.py') }}
 
-    - name: Install
+    - name: Setup pip
       run: |
         python -m pip install --upgrade pip
         pip install --progress-bar off -U setuptools
 
+    - name: Install dependencies with lower bound constraints
+      run: |
+        # Install dependencies with lower bound constraints.
+        pip install alembic==1.5.0 cmaes==0.8.2 packaging==20.0 sqlalchemy==1.3.0
+        pip install plotly==4.9.0 scikit-learn==0.24.2  # optional extras
+      if: ${{ matrix.use-lower-version }}
```

Comment:
I'm not sure if I'm understanding this correctly, but even if a lower version of library A is installed first, if a newer version of A is internally required for the subsequent installation of another library B, won't the newer version of A be installed on its own?

Reply:
This may be a limitation of `pip`. Perhaps something closer to what we want to do is to install a lower limit version after the normal installation?

Reply:
I see, thank you for your insight. Actually it seemingly happened to upgrade `packaging` unexpectedly.
https://github.com/optuna/optuna/actions/runs/3404326039/jobs/5661473599#step:8:171

> Perhaps something closer to what we want to do is to install a lower limit version after the normal installation?

I see. There could be other approach that overwrites `setup.py` (ideally `pyproject.toml`) to pin the versions of packages whose are constrained using a lower-bound. Do you think it is enough to just downgrading after installation as you suggested?

Reply:
24109e1 tried to reinstall packages, it seemingly worked as expected.